prow/github/report/report.go

```diff
@@ -198,65 +220,37 @@ func parseIssueComments(pj prowapi.ProwJob, botName string, ics []github.IssueCo
 	var previousComments []int
 	var latestComment int
 	var entries []string
+	var newComment bool
 	// First accumulate result entries and comment IDs
 	for _, ic := range ics {
 		if ic.User.Login != botName {
 			continue
 		}
-		// Old report comments started with the context. Delete them.
-		// TODO(spxtr): Delete this check a few weeks after this merges.
-		if strings.HasPrefix(ic.Body, pj.Spec.Context) {
-			delete = append(delete, ic.ID)
-		}
 		if !strings.Contains(ic.Body, commentTag) {
 			continue
 		}
 		if latestComment != 0 {
 			previousComments = append(previousComments, latestComment)
 		}
 		latestComment = ic.ID
-		var tracking bool
-		for _, line := range strings.Split(ic.Body, "\n") {
-			line = strings.TrimSpace(line)
-			if strings.HasPrefix(line, "---") {
-				tracking = true
-			} else if len(line) == 0 {
-				tracking = false
-			} else if tracking {
-				entries = append(entries, line)
-			}
+		entries = getOldEntries(ic.Body)
+		// If the first line of the comment does not contain this Job's SHA create a new comment.
+		newComment = !strings.Contains(ic.Body, pj.Spec.Refs.Pulls[0].SHA)
+		if newComment {
+			break
 		}
 	}
 	var newEntries []string
-	// Next decide which entries to keep.
-	for i := range entries {
-		keep := true
-		f1 := strings.Split(entries[i], " | ")
-		for j := range entries {
-			if i == j {
-				continue
-			}
-			f2 := strings.Split(entries[j], " | ")
-			// Use the newer results if there are multiple.
-			if j > i && f2[0] == f1[0] {
-				keep = false
-			}
-		}
-		// Use the current result if there is an old one.
-		if pj.Spec.Context == f1[0] {
-			keep = false
-		}
-		if keep {
-			newEntries = append(newEntries, entries[i])
-		}
+	if !newComment {
```

Comment:
I'm pretty sure this now causes it to throw out table rows for old commits, nice!